lib/pf/UnifiedApi/Controller/Violations.pm

```diff
@@ -0,0 +1,80 @@
+package pf::UnifiedApi::Controller::Violations;
+
+=head1 NAME
+
+pf::UnifiedApi::Controller::Violations -
+
+=cut
+
+=head1 DESCRIPTION
+
+pf::UnifiedApi::Controller::Violations
+
+=cut
+
+use strict;
+use warnings;
+use Mojo::Base 'pf::UnifiedApi::Controller::Crud';
+use pf::violation;
+
+has dal => 'pf::dal::violation';
+has id_key => 'id';
+has resource_id => 'id';
+
+
+sub list {
+    my ($self) = @_;
+    my @violations = pf::violation::violation_view_open_uniq();
+    return $self->render(json => { items => \@violations } ) if scalar @violations > 0 and defined($violations[0]);
+    return $self->render(json => undef);
+}
+
+sub list_by_search {
+    my ($self) = @_;
+    my $search = $self->param('search');
+    return $self->_list_by_mac($search) if pf::util::valid_mac($search);
+    return $self->_list_by_id($search);
+}
+
+sub _list_by_mac {
+    my ($self, $mac) = @_;
+    my @violations = pf::violation::violation_view_desc($mac);
+    return $self->render(json => { items => \@violations }) if scalar @violations > 0 and defined($violations[0]);
+    return $self->render(json => undef);
```

Comment:
@satkunas 

Should render an empty array instead

Reply:
Render an empty array "[]" on a collection, render an empty object "{}" on a row?

Reply:
When a row isn't found it should send a 404 with with a message that says it hasn't been found

When its an empty collection, it should be a 200 with an empty array

Reply:
@julsemaan please confirm again, I updated my last comment above at the same time as your last response.

Reply:
@satkunas it is "{items:[]}" you are right for a collection

Reply:
Adjusted over in https://github.com/inverse-inc/packetfence/pull/2920